Scripts/Runtime/Entities/Lifecycle/Spawning/JobInteraction/EntityDestroyWriter.cs

```diff
@@ -0,0 +1,128 @@
+using Anvil.Unity.DOTS.Data;
+using Anvil.Unity.DOTS.Jobs;
+using System;
+using System.Diagnostics;
+using Unity.Collections;
+using Unity.Entities;
+
+namespace Anvil.Unity.DOTS.Entities
+{
+    /// <summary>
+    /// Helper for queuing up <see cref="Entity"/>s to Destroy in a job.
+    /// Entities will be Destroyed later on via the <see cref="EntityDestroySystem"/>
+    /// </summary>
+    [BurstCompatible]
+    public struct EntityDestroyWriter
```

Comment:
Yeah, it would be nice to have inheritance with structs. Maybe I can do something with some static stateless functions and have all these Writers just define their API so it fits with what the developer is doing but hits the same back logic. Future niceties though.

Reply:
No I mean like this is just a `ParallelTypedStreamWriter` type that anything can use to write to an `UnsafeTypedStream<T>`. Regardless of whether it has to do with Entity destruction or not.

Reply:
Issue created: https://github.com/decline-cookies/anvil-unity-dots/issues/159